src/components/Clock/Clock.tsx

```diff
@@ -0,0 +1,118 @@
+/* eslint-disable no-console */
+import React from 'react';
+
+interface ClockState {
+  currentTime: string,
+  clockName: string,
+  isVisible: boolean,
+}
+
+interface ClockProps {
+  name: string;
+}
+
+export class Clock extends React.Component<ClockProps, ClockState> {
+  state: ClockState = {
+    clockName: this.props.name,
+    currentTime: new Date().toUTCString().slice(-12, -4),
+    isVisible: true,
+  };
+
+  timerID: NodeJS.Timeout | null = null;
+
+  nameTimerID: NodeJS.Timeout | null = null;
+
+  componentDidMount() {
+    this.timerID = setInterval(
+      () => this.updateTime(), 1000,
+    );
+
+    this.nameTimerID = setInterval(
+      () => this.updateName(), 3300,
+    );
+
+    document.addEventListener('contextmenu', this.handleRightClick);
+    document.addEventListener('click', this.handleLeftClick);
```

Comment:
this func is working wrong, i can hidden and show clock only right mouse button, maybe you dont deploy your last version, because code looks right, check it please?